src/__test__/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapLegendVisibilitySettings.test.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { configure, mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import { act } from 'react-dom/test-utils';
+import Adapter from 'enzyme-adapter-react-16';
+import thunk from 'redux-thunk';
+import configureMockStore from 'redux-mock-store';
+import { Radio } from 'antd';
+
+import { UPDATE_CONFIG } from '../../../../../../../redux/actionTypes/componentConfig';
+import HeatmapLegendVisibilitySettings from '../../../../../../../pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapLegendVisibilitySettings';
+
+const initialState = {
+  cellSets: {
+    // cellSets', 'metadataCategorical
+    hierarchy: [
+      {
+        key: 'louvain',
+        children: [
+          {
+            key: 'louvain-0',
+          },
+          {
+            key: 'louvain-1',
+          },
+        ],
+      },
+      {
+        key: 'sample',
+        children: [
+          {
+            key: 'control',
+          },
+          {
+            key: 'treated',
+          },
+        ],
+      },
+    ],
+    properties: {
+      louvain: {
+        type: 'cellSets',
+        name: 'louvain clusters',
+      },
+      'louvain-0': {
+        name: 'louvain 0',
+        cellIds: [5, 6, 7],
+      },
+      'louvain-1': {
+        name: 'louvain 1',
+        cellIds: [1, 2, 3],
+      },
+      sample: {
+        type: 'metadataCategorical',
+        name: 'Sample',
+      },
+      control: {
+        name: 'control',
+        cellIds: [5, 6, 7],
+      },
+      treated: {
+        name: 'treated',
+        cellIds: [1, 2, 3],
+      },
+    },
+  },
+  cellInfo: {},
+  componentConfig: {
+    interactiveHeatmap: {
+      config: {
+        groupedTrack: 'sample',
+        selectedTracks: ['louvain'],
+        legendIsVisible: true,
+      },
+    },
+  },
+};
+
+const mockStore = configureMockStore([thunk]);
+configure({ adapter: new Adapter() });
+
+const eventStub = {
+  stopPropagation: () => { },
+};
+
+describe('HeatmapLegendVisibilitySettings', () => {
+  let component, store, buttons;
+
+  afterEach(() => {
+    component.unmount();
+  });
+
+  beforeEach(() => {
+    store = mockStore({
+      ...initialState,
+    });
+
+    component = mount(
+      <Provider store={store}>
+        {/* ask if experimentId='123' width={200} height={200} are necessary in HeatmapGroupBySettings */}
```

Comment:
to see if they are necessary, go to HeatmapGroupBySettings component and check if in its propTypes these are required. If they are required, then they are necessary, otherwise they are not

Reply:
they are not required then, this question came up becuase I was copying part of the method for testing in HeatmapGroupBySettings, should I delete them there too then of jsut leave it as it is just in case?

Reply:
Leaving commented out code is not a good practice, so I would delete this. Before you delete though, you should check in `HeatmapGroupBySettings` if these parameters are used. Whichever of these parameters will always be needed in `HeatmapGroupBySettings`, you should add them as required.

Reply:
Yes, I was planning to delete before merging.
About the props, they are not being used in `HeatmapGroupBySettings` either as far as I can see.